openmc/mgxs/mdgxs.py

```diff
@@ -0,0 +1,1448 @@
+        The optional user-specified nuclides for which to compute cross
+        sections (e.g., 'U-238', 'O-16'). If by_nuclide is True but nuclides
+        are not specified by the user, all nuclides in the spatial domain
+        are included. This attribute is 'sum' if by_nuclide is false.
+    sparse : bool
+        Whether or not the MDGXS' tallies use SciPy's LIL sparse matrix format
+        for compressed data storage
+    loaded_sp : bool
+        Whether or not a statepoint file has been loaded with tally data
+    derived : bool
+        Whether or not the MDGXS is merged from one or more other MDGXS
+    hdf5_key : str
+        The key used to index multi-group cross sections in an HDF5 data store
+
+    """
+
+    # This is an abstract class which cannot be instantiated
+    __metaclass__ = abc.ABCMeta
+
+    def __init__(self, domain=None, domain_type=None, energy_groups=None,
+                 delayed_groups=None, by_nuclide=False, name=''):
+        super(MDGXS, self).__init__(domain, domain_type, energy_groups,
+                                    by_nuclide, name)
+
+        self._delayed_groups = None
+
+        if delayed_groups is not None:
+            self.delayed_groups = delayed_groups
+
+    def __deepcopy__(self, memo):
+        existing = memo.get(id(self))
+
+        # If this is the first time we have tried to copy this object, copy it
+        if existing is None:
+            clone = type(self).__new__(type(self))
+            clone._name = self.name
+            clone._rxn_type = self.rxn_type
+            clone._by_nuclide = self.by_nuclide
+            clone._nuclides = copy.deepcopy(self._nuclides)
+            clone._domain = self.domain
+            clone._domain_type = self.domain_type
+            clone._energy_groups = copy.deepcopy(self.energy_groups, memo)
+            clone._delayed_groups = copy.deepcopy(self.delayed_groups, memo)
+            clone._tally_trigger = copy.deepcopy(self.tally_trigger, memo)
+            clone._rxn_rate_tally = copy.deepcopy(self._rxn_rate_tally, memo)
+            clone._xs_tally = copy.deepcopy(self._xs_tally, memo)
+            clone._sparse = self.sparse
+            clone._derived = self.derived
+
+            clone._tallies = OrderedDict()
+            for tally_type, tally in self.tallies.items():
+                clone.tallies[tally_type] = copy.deepcopy(tally, memo)
+
+            memo[id(self)] = clone
+
+            return clone
+
+        # If this object has been copied before, return the first copy made
+        else:
+            return existing
+
+    @property
+    def delayed_groups(self):
+        return self._delayed_groups
+
+    @property
+    def num_delayed_groups(self):
+        if self.delayed_groups == None:
+            return 0
+        else:
+            return len(self.delayed_groups)
+
+    @delayed_groups.setter
+    def delayed_groups(self, delayed_groups):
+
+        cv.check_type('delayed groups', delayed_groups, list, int)
```

Comment:
Ok, your concerns make sense to me. Perhaps you can allow the user to pass in an `Iterable` of any type, then convert it to a `list` for storage to the `_delayed_groups` attribute? This would give the user flexibility while allowing you to flexibly do list concatenation under-the-hood.